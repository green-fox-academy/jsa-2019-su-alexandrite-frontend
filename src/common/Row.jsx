import React from 'react';
import { View, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import styles from './styles';

const Row = ({ children, style }) => (
  <View style={{ ...style, ...styles.row }}>
    {children}
  </View>
);


Row.propTypes = {
  children: PropTypes.element,
  style: PropTypes.instanceOf(StyleSheet),
};

Row.defaultProps = {
  children: undefined,
  style: undefined,
};

export default Row;
