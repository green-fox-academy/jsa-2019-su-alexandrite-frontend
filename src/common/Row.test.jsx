import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import Row from './Row';

test('Row render properly', () => {
  const tree = renderer.create(
    <Row />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Row render properly with children', () => {
  const tree = renderer.create(
    <Row><Text>test children</Text></Row>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
