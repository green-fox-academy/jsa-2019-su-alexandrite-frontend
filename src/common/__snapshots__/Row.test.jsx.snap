// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row render properly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
/>
`;

exports[`Row render properly with children 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <Text>
    test children
  </Text>
</View>
`;
